### PR TITLE
Add pagination to contacts and deals

### PR DESCRIPTION
**Pull request by bcody**
This PR adds pagination to our branch of the `AgileCRMWrapper` gem. Specifically:
- Adds a loop to go through more than the default 50 contacts (thanks @nickL )
- Adds same loop for Deals
- Bumps gem version

This functionality is needed for marketing and sales reporting.

### CODE REVIEW

**Comment by bcody** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;EuiG075V&#x2F;758-0-5-agilecrm-gem-improvements"> (0.5) AgileCRM gem improvements</a>

**Comment by coryschires** [comment]
LGTM

File: lib/agilecrm-wrapper/deal.rb [hunk 1]
```diff
@@ -9,7 +9,22 @@ class Deal < Hashie::Mash
...
+        until @no_more_results do
+          response = call_agile_api(@cursor)
+          results.concat(response)
+          @cursor = response.last["cursor"]
+          @no_more_results = @cursor.nil?
+        end
+
+        return results
+      end
+
```
Comment (nickL) [inline lib/agilecrm-wrapper/deal.rb]: :+1:

**Comment by nickL** [comment]
LGTM! Nice work with the refactoring.

File: lib/agilecrm-wrapper/contact.rb [hunk 2]
```diff
@@ -8,7 +8,22 @@ class Contact < Hashie::Mash
 
     class << self
       def all
-        response = AgileCRMWrapper.connection.get('contacts?page_size=5000')
+        results = []
+        @cursor = nil
+
+        until @no_more_results do
+          response = call_agile_api(@cursor)
+          results.concat(response)
```
Comment (georgediaz88) [inline lib/agilecrm-wrapper/contact.rb]: cool use of `concat`

File: lib/agilecrm-wrapper/contact.rb [hunk 3]
```diff
@@ -8,7 +8,22 @@ class Contact < Hashie::Mash
...
+          @cursor = response.last["cursor"]
+          @no_more_results = @cursor.nil?
+        end
+
+        return results
+      end
+
+      def call_agile_api(cursor = nil)
+        response = AgileCRMWrapper.connection.get("contacts?cursor=#{cursor}")
```
Comment (georgediaz88) [inline lib/agilecrm-wrapper/contact.rb]: double quotes ftw!

File: lib/agilecrm-wrapper/deal.rb [hunk 4]
```diff
@@ -9,7 +9,22 @@ class Deal < Hashie::Mash
...
+      def call_agile_api(cursor = nil)
+        response = AgileCRMWrapper.connection.get("opportunity?cursor=#{cursor}")
+
```
Comment (georgediaz88) [inline lib/agilecrm-wrapper/deal.rb]: not necessary now but for later:

Seeing some close similarities in `deal.rb` & `contact.rb` with main difference mainly the api endpoint. You could make your own little DSL (wrapper class) around `AgileCRMWrapper` where you'd have a class with a method you feed the endpoint. For example: `opportunity?cursor=#{cursor}`. 

Then, wrapped around that is the ` until @no_more_results ..` etc. Future food for thoughts.

**Comment by georgediaz88** [comment]
LGTM!! Also, left a future suggestion for _thought_.

**Pull request merged**